aio/aio-proxy/aio_proxy/response/cache/cache.py

```diff
@@ -0,0 +1,52 @@
+import json
+import logging
+from collections.abc import Callable
+
+from aio_proxy.response.cache.redis import RedisClient
+from aio_proxy.response.helpers import hash_string
+
+
+def build_key(key):
+    serialised_key = json.dumps(key.to_dict())
+    return hash_string(serialised_key)
+
+
+def set_cache_value(cache_client, key, value, time_to_live):
+    try:
+        serialised_value = json.dumps(value, default=str)
+        cache_client.set(
+            key,
+            serialised_value,
+            time_to_live,
+        )
+    except Exception as error:
+        logging.info(f"Error while setting value for cache: {error}")
+
+
+def cache_strategy(
+    key,
+    get_response: Callable,
+    should_cache_response: Callable,
+    time_to_live,
+):
+    try:
+        redis_client_cache = RedisClient()
+        # Serialize key object before hashing it
+        request_cache_key = build_key(key)
+        logging.info(f"Request cache key: {request_cache_key}")
+        cached_value = redis_client_cache.get(request_cache_key)
+        if cached_value:
+            return json.loads(cached_value)
+        # Build response
```

Comment:
Comment are really similar to variable name. IMO it indicate that they are either not needed anymore, or you wanna rename those variables

Reply:
Good naming can often replace comment except in really complicated code